integration_tests/test_domestic_migration.py

```diff
@@ -185,50 +183,65 @@ def test_only_living_people_change_households(simulants_on_adjacent_timesteps):
     also that simulants of the same unit have the same details
     """
-    address_cols = [unit_id_col, address_id_col]  # TODO: add state/puma
+    address_cols = [unit_id_col, address_id_col] + state_puma_cols
+    state_puma_map = (
+        sim._data.artifact.load("population.households")
+        .reset_index()
+        .groupby("state")["puma"]
+        .unique()
+    )
     for pop in populations:
         # Check that all simulants in the same unit have the same address
         assert (pop.groupby(unit_id_col)[address_cols].nunique() == 1).all().all()
         # Check that all units have unique addresses (unless it's a PO Box, which has relaxed requirements)
         if address_id_col != "household_details.po_box":
             assert (pop.groupby(address_cols)[unit_id_col].nunique() == 1).all()
-
-    # # TODO implement state/puma checks
-    # # Ensure pumas map to correct state. This will be an imperfect test
-    # # because pumas are only unique within states and so one puma can
-    # # exist in multiple states. Here we only ensure no impossible pumas
-    # # exist in each state
-    # state_puma_map = get_state_puma_map(
-    #     sim._data.artifact.load("population.households").reset_index()
-    # )
-    # for pop in populations:
-    #     for (state, puma) in pop[["state", "puma"]].drop_duplicates().values:
-    #         assert puma in state_puma_map[state]
+        # Check that PUMAs map to correct state. This is an imperfect test
+        # because PUMAs are only unique within states and so one PUMA can
+        # exist in multiple states. Here we only ensure no impossible PUMAs
+        # exist in each state
+        if state_puma_cols:
+            for state_id, puma in pop[state_puma_cols].drop_duplicates().values:
+                assert puma in state_puma_map[state_id]
 
 
 @pytest.mark.parametrize(
-    "unit_id_col, address_id_col",
+    "unit_id_col, address_id_col, state_puma_cols",
     [
-        ("employer_id", "business_details.employer_address_id"),
-        ("household_id", "household_details.address_id"),
+        ("employer_id", "business_details.employer_address_id", []),
+        (
+            "household_id",
+            "household_details.address_id",
+            ["household_details.state_id", "household_details.puma"],
+        ),
     ],
 )
-@pytest.mark.skip(reason="Wait for state/puma to be added (MIC-3728)")
-def test_addresses_during_moves(simulants_on_adjacent_timesteps, unit_id_col, address_id_col):
-    """Check that unit (household and business) address details change after a move."""
-    other_address_cols = []  # TODO: add state/puma
-    address_cols = [address_id_col] + other_address_cols
+def test_addresses_during_moves(
+    simulants_on_adjacent_timesteps, unit_id_col, address_id_col, state_puma_cols
+):
+    """Check that unit (household and business) address details change after a move.
+    NOTE: This test requires a rather large pop size (>~200k) to work since the
```

Comment:
Which part of the test fails? Can we just make it more conservative? For example if it's the 0.001 or 0.95 below let's just change those numbers. I'd prefer not to have a test that always fails with a smaller population.

Reply:
The issue is that the `mask_moved_units` comes back empty. I thought about skipping the tests if it's empty but thought that might cover up bigger problems.

Reply:
I'd skip if it comes back empty! That is more similar to what we do in other tests.

Actually the most similar to what I have done in emigration/immigration tests (which are also very rare events) would be to `continue` the for loop when `mask_moved_units` is empty, but accumulate `mask_moved_units` into a single big Series aggregated across timesteps and then at the end of the test, assert that it's not all False. I'd imagine that would make it work for a ~10x smaller population size, and 20k is what we've been considering our lower bound.